# rubygems_path_support_msys.patch
#
# Target:  lib/rubygems/path_support.rb (pass1 (msys))
# Source:  tebako gem lib/tebako/packager/rubygems_patch.rb RubygemsPatch#rubygems_path_support_patch
#
# Allow only packaged gems (from within the memfs): reset @home to
# Gem.default_dir and keep only @path entries under the memfs mount point
# (A:/__tebako_memfs__), unless TEBAKO_PASS_THROUGH is set.
diff --git a/lib/rubygems/path_support.rb b/lib/rubygems/path_support.rb
index d601e65..34a237b 100644
--- a/lib/rubygems/path_support.rb
+++ b/lib/rubygems/path_support.rb
@@ -31,7 +31,16 @@ class Gem::PathSupport
 
     @home = expand(@home)
 
+  # -- Start of tebako patch --
+    @home = Gem.default_dir unless @home.index("A:/__tebako_memfs__") == 0 unless env["TEBAKO_PASS_THROUGH"]
+# -- End of tebako patch --
     @path = split_gem_path env["GEM_PATH"], @home
+# -- Start of tebako patch --
+    @path.keep_if do |xpath|
+      xpath.index("A:/__tebako_memfs__") == 0
+    end unless env["TEBAKO_PASS_THROUGH"]
+# -- End of tebako patch --
+
 
     @spec_cache_dir = env["GEM_SPEC_CACHE"] || Gem.default_spec_cache_dir
 
